test-utils/test-runner/src/node.rs

```diff
@@ -353,30 +209,29 @@ impl<T: ChainInfo> Node<T> {
 
 	/// Revert all blocks added since creation of the node.
 	pub fn clean(&self) {
-		// if a db path was specified, revert all blocks we've added
-		if let Some(_) = std::env::var("DB_BASE_PATH").ok() {
+		// if the initial block number wasn't 0, then revert blocks
+		if self.initial_block_number != 0u32.into() {
 			let diff = self.client.info().best_number - self.initial_block_number;
 			self.revert_blocks(diff);
 		}
 	}
 
-	/// Performs a runtime upgrade given a wasm blob.
-	pub fn upgrade_runtime(&mut self, wasm: Vec<u8>)
-		where
-			<T::Runtime as frame_system::Config>::Call: From<frame_system::Call<T::Runtime>>
-	{
-		let call = frame_system::Call::set_code(wasm);
-		T::dispatch_with_root(call.into(), self);
+	/// so you've decided to run the test runner as a binary, use this to shutdown gracefully.
+	pub async fn until_shutdown(&self) {
+		let manager = self.task_manager.lock().take();
```

Comment:
using `parking_lot::Mutex` inside of an async function doesn't look like a good idea to me

Reply:
how about `tokio::sync::Mutex`?

Reply:
But is it really needed? Is it used only to allow interior mutability? If so, can you change the signature of the function to `&mut self`?

Reply:
> But is it really needed? 

ahh, you're right, this really shouldn't be called from multiple threads.